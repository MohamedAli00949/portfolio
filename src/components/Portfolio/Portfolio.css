/* start portfolio section */
section#Portfolio p {
  font-size: 20px;
  font-weight: 600;
  color: #09c6f9;
  text-shadow: 0px -1px 0px #000000cf;
  margin-block: 10px;
  text-align: center;
}

section#Portfolio h2 {
  text-align: center;
  font-size: 40px;
  margin-block: 20px;
  color: #09c6f9;
  text-shadow: 2px 1px 3px #000000cf;
}

section#Portfolio ul {
  display: flex;
  align-items: center;
  justify-content: center;
  flex-wrap: wrap;
  padding: 0;
  gap: 10px;
  align-content: center;
}

li.project,
.innerDiv .cover img {
  width: calc(100% / 2.2);
}

.innerDiv {
  width: 100%;
  height: auto;
  position: relative;
  clear: both;
}

.innerDiv .cover {
  position: relative;
}

.innerDiv .cover img {
  opacity: 0;
  min-width: 100%;
}

.innerDiv .cover .main-cover {
  max-width: 100%;
  position: absolute;
  left: 0;
  top: 0;
  right: 0;
  bottom: 0;
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
  border-radius: 2%;
}

.overflow {
  background-color: rgb(37 63 80 / 61%);
  position: absolute;
  left: 0;
  top: 0;
  right: 0;
  bottom: 0;
  z-index: 1;
  opacity: 0;
  visibility: hidden;
  transition: all 0.3s ease;
  -webkit-transition: all 0.3s ease;
  border-radius: 2%;
}

li.project .innerDiv:hover .overflow {
  opacity: 1;
  visibility: visible;
}

li.project .project-details {
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  align-content: center;
  height: 100%;
  position: absolute;
  width: 100%;
  text-align: center;
  bottom: -10px;
  left: 0;
  z-index: 2;
  padding: 20px;
  opacity: 0;
  visibility: hidden;
  transition: all 0.3s ease;
  -webkit-transition: all 0.3s ease;
}

li.project .innerDiv:hover .project-details {
  opacity: 1;
  visibility: visible;
}

.project-details .code {
  margin-block: 10px;
}
/* end portfolio section */

/* start responsive design */
@media screen and (max-width: 768px) {
  li.project,
  .innerDiv .cover img {
    width: 100%;
  }

  ul.skills-container {
    gap: 15px;
  }
}
/* end responsive design */