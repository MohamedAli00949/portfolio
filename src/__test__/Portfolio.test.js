import React, { Suspense } from "react";
import ReactDOM from "react-dom";
import { render } from "@testing-library/react";
import { act } from "react-dom/test-utils";

import Portfolio from "../components/Portfolio/Portfolio";

let container = null;
beforeEach(() => {
    // setup a DOM element as a render target
    container = document.createElement("div");
    container.setAttribute("id", "containeer")
    document.body.appendChild(container);
});

afterEach(() => {
    ReactDOM.unmountComponentAtNode(container);
    container.remove();
    container = null;
});

it("render footer section", async () => {
    await act(async () => {
        await render(<Suspense><Portfolio /></Suspense>)
    });

    container = await document.body;

    expect(container.querySelector("section").getAttribute("id")).toBe("Portfolio");
    expect(container.querySelector("p").textContent).toBe("Selected works");
    expect(container.querySelector("h2").textContent).toBe("Portfolio");
    
    expect(container.querySelectorAll("ul li").length).toBe(6);
    
    expect(container.querySelectorAll("ul li p")[0].textContent).toBe("Phonebook");
    expect(container.querySelectorAll("ul li p")[1].textContent).toBe("Would You Rather...?");
    expect(container.querySelectorAll("ul li p")[2].textContent).toBe("sample twitter");
    expect(container.querySelectorAll("ul li p")[3].textContent).toBe("My reads");
    expect(container.querySelectorAll("ul li p")[4].textContent).toBe("autowats");
    expect(container.querySelectorAll("ul li p")[5].textContent).toBe("landing page");

    expect(container.querySelectorAll("ul li.project a")[0].textContent).toBe("Code in github");
    expect(container.querySelectorAll("ul li.project a")[1].textContent).toBe("Show the demo");

    expect(container.querySelector("ul li.project .overflow")).toBeTruthy();
    expect(container.querySelector("ul li.project .main-cover")).toBeTruthy();

    expect(container).toMatchSnapshot();
})