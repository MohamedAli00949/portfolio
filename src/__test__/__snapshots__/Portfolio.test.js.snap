// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render footer section 1`] = `
<body>
  <div
    id="containeer"
  />
  <div>
    <section
      id="Portfolio"
      style="background: rgb(222, 248, 255); padding: 132px 0px;"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg MuiContainer-fixed css-1ss2ei0-MuiContainer-root"
      >
        <div>
          <p>
            Selected works
          </p>
          <h2>
            Portfolio
          </h2>
          <ul>
            <li
              class="project"
              style="margin-inline: 10px;"
            >
              <div
                class="innerDiv"
              >
                <div
                  class="cover"
                >
                  <img
                    alt="no"
                    height="100%"
                    src="no image.webp"
                    width="100%"
                  />
                  <div
                    class="main-cover"
                    style="background-image: url(phonebook.webp);"
                  />
                </div>
                <div
                  class="overflow"
                />
                <div
                  class="project-details"
                >
                  <p
                    style="font-size: 23px; margin-block: 10px; color: white; text-transform: capitalize;"
                  >
                    Phonebook
                  </p>
                  <div
                    class="code"
                  >
                    <a
                      aria-label="Phonebook github"
                      class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
                      href="https://github.com/MohamedAli00949/phonebook-web-1"
                      style="background: rgb(9, 198, 249);"
                      tabindex="0"
                      target="_blank"
                    >
                      <span
                        class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
                      >
                        <svg
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 1024 1024"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M511.6 76.3C264.3 76.2 64 276.4 64 523.5 64 718.9 189.3 885 363.8 946c23.5 5.9 19.9-10.8 19.9-22.2v-77.5c-135.7 15.9-141.2-73.9-150.3-88.9C215 726 171.5 718 184.5 703c30.9-15.9 62.4 4 98.9 57.9 26.4 39.1 77.9 32.5 104 26 5.7-23.5 17.9-44.5 34.7-60.8-140.6-25.2-199.2-111-199.2-213 0-49.5 16.3-95 48.3-131.7-20.4-60.5 1.9-112.3 4.9-120 58.1-5.2 118.5 41.6 123.2 45.3 33-8.9 70.7-13.6 112.9-13.6 42.4 0 80.2 4.9 113.5 13.9 11.3-8.6 67.3-48.8 121.3-43.9 2.9 7.7 24.7 58.3 5.5 118 32.4 36.8 48.9 82.7 48.9 132.3 0 102.2-59 188.1-200 212.9a127.5 127.5 0 0 1 38.1 91v112.5c.8 9 0 17.9 15 17.9 177.1-59.7 304.6-227 304.6-424.1 0-247.2-200.4-447.3-447.5-447.3z"
                          />
                        </svg>
                      </span>
                      Code in github
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </a>
                  </div>
                  <div
                    class="code"
                  >
                    <a
                      aria-label="Phonebook code"
                      class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
                      href="https://phonebook-web-1.vercel.app/"
                      style="background: rgb(9, 198, 249);"
                      tabindex="0"
                      target="_blank"
                    >
                      <span
                        class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
                      >
                        <svg
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 1024 1024"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M396 512a112 112 0 1 0 224 0 112 112 0 1 0-224 0zm546.2-25.8C847.4 286.5 704.1 186 512 186c-192.2 0-335.4 100.5-430.2 300.3a60.3 60.3 0 0 0 0 51.5C176.6 737.5 319.9 838 512 838c192.2 0 335.4-100.5 430.2-300.3 7.7-16.2 7.7-35 0-51.5zM508 688c-97.2 0-176-78.8-176-176s78.8-176 176-176 176 78.8 176 176-78.8 176-176 176z"
                          />
                        </svg>
                      </span>
                      Show the demo
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </a>
                  </div>
                </div>
              </div>
            </li>
            <li
              class="project"
              style="margin-inline: 10px;"
            >
              <div
                class="innerDiv"
              >
                <div
                  class="cover"
                >
                  <img
                    alt="no"
                    height="100%"
                    src="no image.webp"
                    width="100%"
                  />
                  <div
                    class="main-cover"
                    style="background-image: url(would-you-rather.webp);"
                  />
                </div>
                <div
                  class="overflow"
                />
                <div
                  class="project-details"
                >
                  <p
                    style="font-size: 23px; margin-block: 10px; color: white; text-transform: capitalize;"
                  >
                    Would You Rather...?
                  </p>
                  <div
                    class="code"
                  >
                    <a
                      aria-label="Would You Rather...? github"
                      class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
                      href="https://github.com/MohamedAli00949/Would-You-Rather--ReactND"
                      style="background: rgb(9, 198, 249);"
                      tabindex="0"
                      target="_blank"
                    >
                      <span
                        class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
                      >
                        <svg
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 1024 1024"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M511.6 76.3C264.3 76.2 64 276.4 64 523.5 64 718.9 189.3 885 363.8 946c23.5 5.9 19.9-10.8 19.9-22.2v-77.5c-135.7 15.9-141.2-73.9-150.3-88.9C215 726 171.5 718 184.5 703c30.9-15.9 62.4 4 98.9 57.9 26.4 39.1 77.9 32.5 104 26 5.7-23.5 17.9-44.5 34.7-60.8-140.6-25.2-199.2-111-199.2-213 0-49.5 16.3-95 48.3-131.7-20.4-60.5 1.9-112.3 4.9-120 58.1-5.2 118.5 41.6 123.2 45.3 33-8.9 70.7-13.6 112.9-13.6 42.4 0 80.2 4.9 113.5 13.9 11.3-8.6 67.3-48.8 121.3-43.9 2.9 7.7 24.7 58.3 5.5 118 32.4 36.8 48.9 82.7 48.9 132.3 0 102.2-59 188.1-200 212.9a127.5 127.5 0 0 1 38.1 91v112.5c.8 9 0 17.9 15 17.9 177.1-59.7 304.6-227 304.6-424.1 0-247.2-200.4-447.3-447.5-447.3z"
                          />
                        </svg>
                      </span>
                      Code in github
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </a>
                  </div>
                  <div
                    class="code"
                  >
                    <a
                      aria-label="Would You Rather...? code"
                      class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
                      href="https://would-you-rather-react-nd.vercel.app/"
                      style="background: rgb(9, 198, 249);"
                      tabindex="0"
                      target="_blank"
                    >
                      <span
                        class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
                      >
                        <svg
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 1024 1024"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M396 512a112 112 0 1 0 224 0 112 112 0 1 0-224 0zm546.2-25.8C847.4 286.5 704.1 186 512 186c-192.2 0-335.4 100.5-430.2 300.3a60.3 60.3 0 0 0 0 51.5C176.6 737.5 319.9 838 512 838c192.2 0 335.4-100.5 430.2-300.3 7.7-16.2 7.7-35 0-51.5zM508 688c-97.2 0-176-78.8-176-176s78.8-176 176-176 176 78.8 176 176-78.8 176-176 176z"
                          />
                        </svg>
                      </span>
                      Show the demo
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </a>
                  </div>
                </div>
              </div>
            </li>
            <li
              class="project"
              style="margin-inline: 10px;"
            >
              <div
                class="innerDiv"
              >
                <div
                  class="cover"
                >
                  <img
                    alt="no"
                    height="100%"
                    src="no image.webp"
                    width="100%"
                  />
                  <div
                    class="main-cover"
                    style="background-image: url(sample-twitter.webp);"
                  />
                </div>
                <div
                  class="overflow"
                />
                <div
                  class="project-details"
                >
                  <p
                    style="font-size: 23px; margin-block: 10px; color: white; text-transform: capitalize;"
                  >
                    sample twitter
                  </p>
                  <div
                    class="code"
                  >
                    <a
                      aria-label="sample twitter github"
                      class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
                      href="https://github.com/MohamedAli00949/simple_twitter"
                      style="background: rgb(9, 198, 249);"
                      tabindex="0"
                      target="_blank"
                    >
                      <span
                        class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
                      >
                        <svg
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 1024 1024"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M511.6 76.3C264.3 76.2 64 276.4 64 523.5 64 718.9 189.3 885 363.8 946c23.5 5.9 19.9-10.8 19.9-22.2v-77.5c-135.7 15.9-141.2-73.9-150.3-88.9C215 726 171.5 718 184.5 703c30.9-15.9 62.4 4 98.9 57.9 26.4 39.1 77.9 32.5 104 26 5.7-23.5 17.9-44.5 34.7-60.8-140.6-25.2-199.2-111-199.2-213 0-49.5 16.3-95 48.3-131.7-20.4-60.5 1.9-112.3 4.9-120 58.1-5.2 118.5 41.6 123.2 45.3 33-8.9 70.7-13.6 112.9-13.6 42.4 0 80.2 4.9 113.5 13.9 11.3-8.6 67.3-48.8 121.3-43.9 2.9 7.7 24.7 58.3 5.5 118 32.4 36.8 48.9 82.7 48.9 132.3 0 102.2-59 188.1-200 212.9a127.5 127.5 0 0 1 38.1 91v112.5c.8 9 0 17.9 15 17.9 177.1-59.7 304.6-227 304.6-424.1 0-247.2-200.4-447.3-447.5-447.3z"
                          />
                        </svg>
                      </span>
                      Code in github
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </a>
                  </div>
                  <div
                    class="code"
                  >
                    <a
                      aria-label="sample twitter code"
                      class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
                      href="https://sample-twitter.netlify.app/"
                      style="background: rgb(9, 198, 249);"
                      tabindex="0"
                      target="_blank"
                    >
                      <span
                        class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
                      >
                        <svg
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 1024 1024"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M396 512a112 112 0 1 0 224 0 112 112 0 1 0-224 0zm546.2-25.8C847.4 286.5 704.1 186 512 186c-192.2 0-335.4 100.5-430.2 300.3a60.3 60.3 0 0 0 0 51.5C176.6 737.5 319.9 838 512 838c192.2 0 335.4-100.5 430.2-300.3 7.7-16.2 7.7-35 0-51.5zM508 688c-97.2 0-176-78.8-176-176s78.8-176 176-176 176 78.8 176 176-78.8 176-176 176z"
                          />
                        </svg>
                      </span>
                      Show the demo
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </a>
                  </div>
                </div>
              </div>
            </li>
            <li
              class="project"
              style="margin-inline: 10px;"
            >
              <div
                class="innerDiv"
              >
                <div
                  class="cover"
                >
                  <img
                    alt="no"
                    height="100%"
                    src="no image.webp"
                    width="100%"
                  />
                  <div
                    class="main-cover"
                    style="background-image: url(myreads.webp);"
                  />
                </div>
                <div
                  class="overflow"
                />
                <div
                  class="project-details"
                >
                  <p
                    style="font-size: 23px; margin-block: 10px; color: white; text-transform: capitalize;"
                  >
                    My reads
                  </p>
                  <div
                    class="code"
                  >
                    <a
                      aria-label="My reads github"
                      class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
                      href="https://github.com/MohamedAli00949/reactnd-project-myreads"
                      style="background: rgb(9, 198, 249);"
                      tabindex="0"
                      target="_blank"
                    >
                      <span
                        class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
                      >
                        <svg
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 1024 1024"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M511.6 76.3C264.3 76.2 64 276.4 64 523.5 64 718.9 189.3 885 363.8 946c23.5 5.9 19.9-10.8 19.9-22.2v-77.5c-135.7 15.9-141.2-73.9-150.3-88.9C215 726 171.5 718 184.5 703c30.9-15.9 62.4 4 98.9 57.9 26.4 39.1 77.9 32.5 104 26 5.7-23.5 17.9-44.5 34.7-60.8-140.6-25.2-199.2-111-199.2-213 0-49.5 16.3-95 48.3-131.7-20.4-60.5 1.9-112.3 4.9-120 58.1-5.2 118.5 41.6 123.2 45.3 33-8.9 70.7-13.6 112.9-13.6 42.4 0 80.2 4.9 113.5 13.9 11.3-8.6 67.3-48.8 121.3-43.9 2.9 7.7 24.7 58.3 5.5 118 32.4 36.8 48.9 82.7 48.9 132.3 0 102.2-59 188.1-200 212.9a127.5 127.5 0 0 1 38.1 91v112.5c.8 9 0 17.9 15 17.9 177.1-59.7 304.6-227 304.6-424.1 0-247.2-200.4-447.3-447.5-447.3z"
                          />
                        </svg>
                      </span>
                      Code in github
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </a>
                  </div>
                  <div
                    class="code"
                  >
                    <a
                      aria-label="My reads code"
                      class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
                      href="http://reactnd-project-myreads-ivory.vercel.app/"
                      style="background: rgb(9, 198, 249);"
                      tabindex="0"
                      target="_blank"
                    >
                      <span
                        class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
                      >
                        <svg
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 1024 1024"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M396 512a112 112 0 1 0 224 0 112 112 0 1 0-224 0zm546.2-25.8C847.4 286.5 704.1 186 512 186c-192.2 0-335.4 100.5-430.2 300.3a60.3 60.3 0 0 0 0 51.5C176.6 737.5 319.9 838 512 838c192.2 0 335.4-100.5 430.2-300.3 7.7-16.2 7.7-35 0-51.5zM508 688c-97.2 0-176-78.8-176-176s78.8-176 176-176 176 78.8 176 176-78.8 176-176 176z"
                          />
                        </svg>
                      </span>
                      Show the demo
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </a>
                  </div>
                </div>
              </div>
            </li>
            <li
              class="project"
              style="margin-inline: 10px;"
            >
              <div
                class="innerDiv"
              >
                <div
                  class="cover"
                >
                  <img
                    alt="no"
                    height="100%"
                    src="no image.webp"
                    width="100%"
                  />
                  <div
                    class="main-cover"
                    style="background-image: url(autowats.webp);"
                  />
                </div>
                <div
                  class="overflow"
                />
                <div
                  class="project-details"
                >
                  <p
                    style="font-size: 23px; margin-block: 10px; color: white; text-transform: capitalize;"
                  >
                    autowats
                  </p>
                  <div
                    class="code"
                  >
                    <a
                      aria-label="autowats github"
                      class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
                      href="https://github.com/MohamedAli00949/autowarts"
                      style="background: rgb(9, 198, 249);"
                      tabindex="0"
                      target="_blank"
                    >
                      <span
                        class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
                      >
                        <svg
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 1024 1024"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M511.6 76.3C264.3 76.2 64 276.4 64 523.5 64 718.9 189.3 885 363.8 946c23.5 5.9 19.9-10.8 19.9-22.2v-77.5c-135.7 15.9-141.2-73.9-150.3-88.9C215 726 171.5 718 184.5 703c30.9-15.9 62.4 4 98.9 57.9 26.4 39.1 77.9 32.5 104 26 5.7-23.5 17.9-44.5 34.7-60.8-140.6-25.2-199.2-111-199.2-213 0-49.5 16.3-95 48.3-131.7-20.4-60.5 1.9-112.3 4.9-120 58.1-5.2 118.5 41.6 123.2 45.3 33-8.9 70.7-13.6 112.9-13.6 42.4 0 80.2 4.9 113.5 13.9 11.3-8.6 67.3-48.8 121.3-43.9 2.9 7.7 24.7 58.3 5.5 118 32.4 36.8 48.9 82.7 48.9 132.3 0 102.2-59 188.1-200 212.9a127.5 127.5 0 0 1 38.1 91v112.5c.8 9 0 17.9 15 17.9 177.1-59.7 304.6-227 304.6-424.1 0-247.2-200.4-447.3-447.5-447.3z"
                          />
                        </svg>
                      </span>
                      Code in github
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </a>
                  </div>
                  <div
                    class="code"
                  >
                    <a
                      aria-label="autowats code"
                      class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
                      href="https://mohamedali00949.github.io/autowarts/"
                      style="background: rgb(9, 198, 249);"
                      tabindex="0"
                      target="_blank"
                    >
                      <span
                        class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
                      >
                        <svg
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 1024 1024"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M396 512a112 112 0 1 0 224 0 112 112 0 1 0-224 0zm546.2-25.8C847.4 286.5 704.1 186 512 186c-192.2 0-335.4 100.5-430.2 300.3a60.3 60.3 0 0 0 0 51.5C176.6 737.5 319.9 838 512 838c192.2 0 335.4-100.5 430.2-300.3 7.7-16.2 7.7-35 0-51.5zM508 688c-97.2 0-176-78.8-176-176s78.8-176 176-176 176 78.8 176 176-78.8 176-176 176z"
                          />
                        </svg>
                      </span>
                      Show the demo
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </a>
                  </div>
                </div>
              </div>
            </li>
            <li
              class="project"
              style="margin-inline: 10px;"
            >
              <div
                class="innerDiv"
              >
                <div
                  class="cover"
                >
                  <img
                    alt="no"
                    height="100%"
                    src="no image.webp"
                    width="100%"
                  />
                  <div
                    class="main-cover"
                    style="background-image: url(landing-page.webp);"
                  />
                </div>
                <div
                  class="overflow"
                />
                <div
                  class="project-details"
                >
                  <p
                    style="font-size: 23px; margin-block: 10px; color: white; text-transform: capitalize;"
                  >
                    landing page
                  </p>
                  <div
                    class="code"
                  >
                    <a
                      aria-label="landing page github"
                      class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
                      href="https://github.com/MohamedAli00949/Udacity-front-end-Nanodegee-Landing-Page"
                      style="background: rgb(9, 198, 249);"
                      tabindex="0"
                      target="_blank"
                    >
                      <span
                        class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
                      >
                        <svg
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 1024 1024"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M511.6 76.3C264.3 76.2 64 276.4 64 523.5 64 718.9 189.3 885 363.8 946c23.5 5.9 19.9-10.8 19.9-22.2v-77.5c-135.7 15.9-141.2-73.9-150.3-88.9C215 726 171.5 718 184.5 703c30.9-15.9 62.4 4 98.9 57.9 26.4 39.1 77.9 32.5 104 26 5.7-23.5 17.9-44.5 34.7-60.8-140.6-25.2-199.2-111-199.2-213 0-49.5 16.3-95 48.3-131.7-20.4-60.5 1.9-112.3 4.9-120 58.1-5.2 118.5 41.6 123.2 45.3 33-8.9 70.7-13.6 112.9-13.6 42.4 0 80.2 4.9 113.5 13.9 11.3-8.6 67.3-48.8 121.3-43.9 2.9 7.7 24.7 58.3 5.5 118 32.4 36.8 48.9 82.7 48.9 132.3 0 102.2-59 188.1-200 212.9a127.5 127.5 0 0 1 38.1 91v112.5c.8 9 0 17.9 15 17.9 177.1-59.7 304.6-227 304.6-424.1 0-247.2-200.4-447.3-447.5-447.3z"
                          />
                        </svg>
                      </span>
                      Code in github
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </a>
                  </div>
                  <div
                    class="code"
                  >
                    <a
                      aria-label="landing page code"
                      class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
                      href="https://mohamedali00949.github.io/Udacity-front-end-Nanodegee-Landing-Page/"
                      style="background: rgb(9, 198, 249);"
                      tabindex="0"
                      target="_blank"
                    >
                      <span
                        class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
                      >
                        <svg
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 1024 1024"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M396 512a112 112 0 1 0 224 0 112 112 0 1 0-224 0zm546.2-25.8C847.4 286.5 704.1 186 512 186c-192.2 0-335.4 100.5-430.2 300.3a60.3 60.3 0 0 0 0 51.5C176.6 737.5 319.9 838 512 838c192.2 0 335.4-100.5 430.2-300.3 7.7-16.2 7.7-35 0-51.5zM508 688c-97.2 0-176-78.8-176-176s78.8-176 176-176 176 78.8 176 176-78.8 176-176 176z"
                          />
                        </svg>
                      </span>
                      Show the demo
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </a>
                  </div>
                </div>
              </div>
            </li>
          </ul>
        </div>
      </div>
    </section>
  </div>
</body>
`;
